tsdb/engine/tsm1/engine.go

```diff
@@ -150,6 +154,7 @@ func (e *Engine) Close() error {
 	// Lock now and close everything else down.
 	e.mu.Lock()
 	defer e.mu.Unlock()
+	e.done = nil // Ensures that the channel will not be closed again.
```

Comment:
These should be the cases I discussed above. They won't be in their `select` loops if `w.wg` is done, since the `wg` can't be done until anything receiving on `done` has returned.

Maybe I should change the approach to something clearer?